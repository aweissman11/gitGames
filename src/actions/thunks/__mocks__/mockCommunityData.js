export const mockCommunityData = {
  "followers": {
    "totalCount": 14
  },
  "following": {
    "totalCount": 5
  },
  "starredRepositories": {
    "totalCount": 16
  },
  "repositories": {
    "totalCount": 64
  },
  "repositoriesContributedTo": {
    "totalCount": 9
  },
  "watching": {
    "totalCount": 1
  },
  "issues": {
    "totalCount": 92
  },
  "issueComments": {
    "totalCount": 35
  },
  "pullRequests": {
    "totalCount": 233
  }
}