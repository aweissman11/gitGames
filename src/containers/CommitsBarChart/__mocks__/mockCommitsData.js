export const mockCommitsData = [
  {
    "firstDay": "2017-12-31",
    "contributionDays": [
      {
        "date": "2017-12-31",
        "contributionCount": 0
      },
      {
        "date": "2018-01-01",
        "contributionCount": 0
      },
      {
        "date": "2018-01-02",
        "contributionCount": 0
      },
      {
        "date": "2018-01-03",
        "contributionCount": 0
      },
      {
        "date": "2018-01-04",
        "contributionCount": 0
      },
      {
        "date": "2018-01-05",
        "contributionCount": 0
      },
      {
        "date": "2018-01-06",
        "contributionCount": 0
      }
    ],
    "averageCommits": 0
  },
  {
    "firstDay": "2018-01-07",
    "contributionDays": [
      {
        "date": "2018-01-07",
        "contributionCount": 0
      },
      {
        "date": "2018-01-08",
        "contributionCount": 0
      },
      {
        "date": "2018-01-09",
        "contributionCount": 0
      },
      {
        "date": "2018-01-10",
        "contributionCount": 0
      },
      {
        "date": "2018-01-11",
        "contributionCount": 0
      },
      {
        "date": "2018-01-12",
        "contributionCount": 0
      },
      {
        "date": "2018-01-13",
        "contributionCount": 0
      }
    ],
    "averageCommits": 0
  },
  {
    "firstDay": "2018-01-14",
    "contributionDays": [
      {
        "date": "2018-01-14",
        "contributionCount": 0
      },
      {
        "date": "2018-01-15",
        "contributionCount": 0
      },
      {
        "date": "2018-01-16",
        "contributionCount": 0
      },
      {
        "date": "2018-01-17",
        "contributionCount": 0
      },
      {
        "date": "2018-01-18",
        "contributionCount": 0
      },
      {
        "date": "2018-01-19",
        "contributionCount": 0
      },
      {
        "date": "2018-01-20",
        "contributionCount": 0
      }
    ],
    "averageCommits": 0
  }
]