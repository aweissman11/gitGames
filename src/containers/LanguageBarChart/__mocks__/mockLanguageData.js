export const mockLanguageData = {
  "Overall": {
    "Ruby": 0.3262813173751577,
    "CSS": 0.027094665090688645,
    "HTML": 0.08554854057929996,
    "Roff": 0.0013872963043256535,
    "JavaScript": 0.5595484354944749,
    "CoffeeScript": 0.00013974515605317292
  },
  "Repositories": [
    {
      "Ruby": 1,
      "name": "credit_check"
    },
    {
      "Ruby": 1,
      "name": "sorting_cards"
    },
    {
      "Ruby": 1,
      "name": "morse_translator"
    },
    {
      "Ruby": 1,
      "name": "enigma"
    },
    {
      "Ruby": 1,
      "name": "scrabble"
    },
    {
      "Ruby": 1,
      "name": "wizarding_bank"
    },
    {
      "Ruby": 0.09082092918358142,
      "CSS": 0.7837843933243122,
      "HTML": 0.12539467749210645,
      "name": "denver_public_library"
    },
    {
      "Ruby": 1,
      "name": "reunion"
    },
    {
      "Ruby": 1,
      "name": "pantry"
    },
    {
      "Ruby": 1,
      "name": "Award_Show"
    },
    {
      "Ruby": 1,
      "name": "SuperFizzBuzz"
    },
    {
      "Ruby": 1,
      "name": "municipal-budget"
    },
    {
      "Ruby": 1,
      "name": "credit_check2"
    },
    {
      "Ruby": 1,
      "name": "sorting_cards2"
    },
    {
      "Ruby": 1,
      "name": "dog_party"
    },
    {
      "Ruby": 1,
      "name": "binary_translator"
    },
    {
      "Ruby": 1,
      "name": "night_writer"
    },
    {
      "Ruby": 1,
      "name": "potluck"
    },
    {
      "Ruby": 1,
      "name": "connect_four"
    },
    {
      "Ruby": 1,
      "name": "world_cup"
    },
    {
      "Ruby": 1,
      "name": "black_thursday"
    },
    {
      "Ruby": 1,
      "name": "pantry-1"
    },
    {
      "Ruby": 1,
      "name": "market"
    },
    {
      "Roff": 0.400765306122449,
      "JavaScript": 0.5700892857142857,
      "Ruby": 0.029145408163265305,
      "name": "portfolios"
    },
    {
      "Ruby": 0.8176470588235294,
      "HTML": 0.1715170278637771,
      "CSS": 0.010835913312693499,
      "name": "personal_site"
    },
    {
      "HTML": 0.5896462122726365,
      "CSS": 0.4103537877273636,
      "name": "static_challenges"
    },
    {
      "Ruby": 0.710048760870658,
      "HTML": 0.28995123912934195,
      "name": "little-shop-redux"
    },
    {
      "Ruby": 0.859143593045718,
      "JavaScript": 0.015507619660871432,
      "CoffeeScript": 0.008491629104958145,
      "CSS": 0.0077940545181369395,
      "HTML": 0.10906310367031552,
      "name": "activerecord-obstacle-course"
    },
    {
      "Ruby": 0.9851824181297473,
      "HTML": 0.01481758187025277,
      "name": "laugh_tracks"
    },
    {
      "HTML": 0.5709939148073022,
      "CSS": 0.4290060851926978,
      "name": "set-list-song-index-B2"
    },
    {
      "Ruby": 0.7972954105750244,
      "JavaScript": 0.02259808125735485,
      "CSS": 0.045615014859285555,
      "HTML": 0.13449149330833515,
      "name": "jukebox"
    },
    {
      "Ruby": 0.7548183647642165,
      "JavaScript": 0.0064106553805937634,
      "CSS": 0.06989481546026016,
      "HTML": 0.16887616439492956,
      "name": "job-tracker"
    },
    {
      "Ruby": 0.8147995739736639,
      "JavaScript": 0.027425445391169635,
      "CSS": 0.017162083656080556,
      "HTML": 0.14061289697908597,
      "name": "apollo_14"
    },
    {
      "Ruby": 0.830796071027412,
      "JavaScript": 0.027077408407618957,
      "HTML": 0.14212652056496905,
      "name": "blogger"
    },
    {
      "Ruby": 0.8214013783409149,
      "JavaScript": 0.02211983366197458,
      "CSS": 0.013841978875851702,
      "HTML": 0.14263680912125887,
      "name": "school"
    },
    {
      "Ruby": 0.7420724875583001,
      "JavaScript": 0.014760676411579248,
      "CSS": 0.08808202402355454,
      "HTML": 0.1550848120065661,
      "name": "Plant_Pal"
    },
    {
      "Ruby": 0.8214620222113197,
      "JavaScript": 0.02526649123589492,
      "CSS": 0.015811069979037508,
      "HTML": 0.13746041657374783,
      "name": "vending-machine-tracker"
    },
    {
      "Ruby": 0.7645979514476974,
      "JavaScript": 0.006018630534720542,
      "CSS": 0.001986047261875958,
      "HTML": 0.2273973707557061,
      "name": "bike_share"
    },
    {
      "Ruby": 0.2317893931450756,
      "JavaScript": 0.0053511294992230705,
      "CSS": 0.1674861026491161,
      "HTML": 0.5953733747065852,
      "name": "jukebox_1806_final"
    },
    {
      "Ruby": 0.9975892572521518,
      "HTML": 0.0024107427478482625,
      "name": "rales_engine"
    },
    {
      "Ruby": 0.8716763721967538,
      "JavaScript": 0.021323256102093915,
      "CSS": 0.01777970511708586,
      "HTML": 0.08922066658406641,
      "name": "battleshift"
    },
    {
      "Ruby": 0.8309996290801187,
      "JavaScript": 0.014790430267062315,
      "CSS": 0.0019241468842729971,
      "HTML": 0.152285793768546,
      "name": "module_3_diagnostic"
    },
    {
      "Ruby": 0.7645979514476974,
      "JavaScript": 0.006018630534720542,
      "CSS": 0.001986047261875958,
      "HTML": 0.2273973707557061,
      "name": "bike_share"
    },
    {
      "Ruby": 0.6937520985613554,
      "JavaScript": 0.02866957667174626,
      "CSS": 0.014890099956091639,
      "HTML": 0.2626882248108066,
      "name": "Trelora"
    },
    {
      "Ruby": 0.8353935362071291,
      "JavaScript": 0.014339012001618195,
      "CSS": 0.00186542005663685,
      "HTML": 0.1484020317346159,
      "name": "module_3_diagnostic-1"
    },
    {
      "Ruby": 0.8226148265884174,
      "JavaScript": 0.023087582018991727,
      "CSS": 0.02871174145168521,
      "HTML": 0.12558584994090558,
      "name": "module_3_final_assessment"
    },
    {
      "Ruby": 0.8622623244260169,
      "JavaScript": 0.021589435423478055,
      "CSS": 0.012950008219328207,
      "HTML": 0.10319823193117683,
      "name": "Congress"
    },
    {
      "HTML": 0.2774971484438651,
      "JavaScript": 0.35929607299983707,
      "CSS": 0.36320677855629785,
      "name": "number_guesser"
    },
    {
      "JavaScript": 1,
      "name": "sorting_suite"
    },
    {
      "HTML": 0.24405682937121959,
      "JavaScript": 0.5078773385848924,
      "CSS": 0.24806583204388802,
      "name": "persephone"
    },
    {
      "HTML": 0.20968821729150727,
      "JavaScript": 0.6050114766641163,
      "CSS": 0.18530030604437644,
      "name": "RepresentativeTracker"
    },
    {
      "JavaScript": 0.9961325277139719,
      "HTML": 0.0018018242950830587,
      "CSS": 0.0020656479909451045,
      "name": "Play-FE"
    },
    {
      "JavaScript": 1,
      "name": "play-be"
    },
    {
      "JavaScript": 0.4117534088726714,
      "CSS": 0.5882465911273286,
      "name": "word_watch"
    },
    {
      "Ruby": 0.9952243754193472,
      "HTML": 0.0047756245806528,
      "name": "GitGames-BackEnd"
    },
    {
      "HTML": 0.0266348273328435,
      "CSS": 0.1057339060645453,
      "JavaScript": 0.8676312666026113,
      "name": "gitGames"
    },
    {
      "Ruby": 0.8175188582652485,
      "JavaScript": 0.024217982740246984,
      "CSS": 0.01481496959692417,
      "HTML": 0.14344818939758028,
      "name": "promises"
    },
    {
      "Ruby": 0.9921272650378997,
      "HTML": 0.007872734962100263,
      "name": "gitGames_be_messaging"
    }
  ]
}