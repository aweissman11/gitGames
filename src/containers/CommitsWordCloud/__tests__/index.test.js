import React from 'react';
import { shallow } from 'enzyme';

import randomColor from 'randomcolor';


import { CommitsWordCloud } from '../index';
import { mapStateToProps } from '../index';

import { mockCloudData } from '../__mocks__/mockCloudData';

// jest.mock('randomcolor', () => ({
//   randomColor: () => ({
//     error: {
//       message: 'error message'
//     }
//   })
// }));

describe('CommitsWordCloud', () => {

  let wrapper;
  // const mockForceUpdate = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<CommitsWordCloud
                        // onClick={mockForceUpdate}
                        cloudData={mockCloudData}
                        loadingWordCloud={{
                          loadingWordCloud: false,
                          message: 'not loading'
                        }}
                      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should forceUpdate', () => {
    const mockForceUpdate = jest.fn();
    wrapper.instance().forceUpdate = mockForceUpdate    
    wrapper.find('.tag-cloud').simulate('click')
    expect(mockForceUpdate).toHaveBeenCalled()
  })
  
  // it('should get a randomColor', () => {
  //   // randomColor = jest.fn();
  //   wrapper.find('.tag-cloud').simulate('click')
  //   expect(randomColor).toHaveBeenCalled()
  // })

  describe('getFontSize', () => {
    let styles;
    
    beforeEach(() => {
      styles = {
        small: {
          fontSize: 14,
          opacity: 0.7
        },
        medium: {
          fontSize: 22,
        },
        big: {
          fontSize: 30,
        },
        huge: {
          fontSize: 50,
          fontWeight: 'bold'
        }
      }
    })
    
    it('should get small font', () => {
      const expected = styles.small
      const result = wrapper.instance().getFontSize(1)
      expect(result).toEqual(expected)
    })

    it('should get medium font', () => {
      const expected = styles.medium
      const result = wrapper.instance().getFontSize(5)
      expect(result).toEqual(expected)
    })

    it('should get big font', () => {
      const expected = styles.big
      const result = wrapper.instance().getFontSize(15)
      expect(result).toEqual(expected)
    })

    it('should get huge font', () => {
      const expected = styles.huge
      const result = wrapper.instance().getFontSize(25)
      expect(result).toEqual(expected)
    })
  })

  describe('mapStateToProps', () => {
    it('should map the campslist state to props', () => {
      const expected = mockCloudData
      const mockState = {
        cloudData: mockCloudData
      }
      
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps.cloudData).toEqual(expected)
    })

    it('should map the loading state to props', () => {
      const expected = {
        loadingWordCloud: true,
        message: 'loading right now'
      }
      const mockState = {
        loadingWordCloud: expected
      }
      
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps.loadingWordCloud).toEqual(expected)
    })
  })
  
})
