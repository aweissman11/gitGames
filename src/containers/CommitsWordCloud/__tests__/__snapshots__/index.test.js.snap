// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitsWordCloud should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommitsWordCloud
    cloudData={
      Object {
        "activity_index_spec": 1,
        "admin": 7,
        "completes": 2,
        "edit\\"": 1,
        "readme": 9,
        "removes": 12,
        "testing": 2,
        "update": 19,
        "vulnerability": 1,
      }
    }
    loadingWordCloud={
      Object {
        "loadingWordCloud": false,
        "message": "not loading",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="tag-container"
      >
        <TagCloud
          className="tag-cloud"
          onClick={[Function]}
          random={[Function]}
          rotate={0}
          spiral="archimedean"
          style={
            Object {
              "color": [Function],
              "fontFamily": "sans-serif",
              "fontSize": 22,
              "height": "100%",
              "padding": 5,
              "rotate": 90,
              "width": "100%",
            }
          }
        >
          <CloudItem
            instances={19}
            style={
              Object {
                "fontSize": 30,
              }
            }
            text="update"
          >
            update
          </CloudItem>
          <CloudItem
            instances={9}
            style={
              Object {
                "fontSize": 30,
              }
            }
            text="readme"
          >
            readme
          </CloudItem>
          <CloudItem
            instances={12}
            style={
              Object {
                "fontSize": 30,
              }
            }
            text="removes"
          >
            removes
          </CloudItem>
          <CloudItem
            instances={1}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="vulnerability"
          >
            vulnerability
          </CloudItem>
          <CloudItem
            instances={2}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="completes"
          >
            completes
          </CloudItem>
          <CloudItem
            instances={7}
            style={
              Object {
                "fontSize": 22,
              }
            }
            text="admin"
          >
            admin
          </CloudItem>
          <CloudItem
            instances={2}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="testing"
          >
            testing
          </CloudItem>
          <CloudItem
            instances={1}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="activity_index_spec"
          >
            activity_index_spec
          </CloudItem>
          <CloudItem
            instances={1}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="edit\\""
          >
            edit"
          </CloudItem>
        </TagCloud>
      </div>,
      "className": "commits-word-cloud",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TagCloud
          className="tag-cloud"
          onClick={[Function]}
          random={[Function]}
          rotate={0}
          spiral="archimedean"
          style={
            Object {
              "color": [Function],
              "fontFamily": "sans-serif",
              "fontSize": 22,
              "height": "100%",
              "padding": 5,
              "rotate": 90,
              "width": "100%",
            }
          }
        >
          <CloudItem
            instances={19}
            style={
              Object {
                "fontSize": 30,
              }
            }
            text="update"
          >
            update
          </CloudItem>
          <CloudItem
            instances={9}
            style={
              Object {
                "fontSize": 30,
              }
            }
            text="readme"
          >
            readme
          </CloudItem>
          <CloudItem
            instances={12}
            style={
              Object {
                "fontSize": 30,
              }
            }
            text="removes"
          >
            removes
          </CloudItem>
          <CloudItem
            instances={1}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="vulnerability"
          >
            vulnerability
          </CloudItem>
          <CloudItem
            instances={2}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="completes"
          >
            completes
          </CloudItem>
          <CloudItem
            instances={7}
            style={
              Object {
                "fontSize": 22,
              }
            }
            text="admin"
          >
            admin
          </CloudItem>
          <CloudItem
            instances={2}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="testing"
          >
            testing
          </CloudItem>
          <CloudItem
            instances={1}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="activity_index_spec"
          >
            activity_index_spec
          </CloudItem>
          <CloudItem
            instances={1}
            style={
              Object {
                "fontSize": 14,
                "opacity": 0.7,
              }
            }
            text="edit\\""
          >
            edit"
          </CloudItem>
        </TagCloud>,
        "className": "tag-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <CloudItem
              instances={19}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="update"
            >
              update
            </CloudItem>,
            <CloudItem
              instances={9}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="readme"
            >
              readme
            </CloudItem>,
            <CloudItem
              instances={12}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="removes"
            >
              removes
            </CloudItem>,
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="vulnerability"
            >
              vulnerability
            </CloudItem>,
            <CloudItem
              instances={2}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="completes"
            >
              completes
            </CloudItem>,
            <CloudItem
              instances={7}
              style={
                Object {
                  "fontSize": 22,
                }
              }
              text="admin"
            >
              admin
            </CloudItem>,
            <CloudItem
              instances={2}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="testing"
            >
              testing
            </CloudItem>,
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="activity_index_spec"
            >
              activity_index_spec
            </CloudItem>,
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="edit\\""
            >
              edit"
            </CloudItem>,
          ],
          "className": "tag-cloud",
          "onClick": [Function],
          "random": [Function],
          "rotate": 0,
          "spiral": "archimedean",
          "style": Object {
            "color": [Function],
            "fontFamily": "sans-serif",
            "fontSize": 22,
            "height": "100%",
            "padding": 5,
            "rotate": 90,
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "update",
            "nodeType": "function",
            "props": Object {
              "children": "update",
              "instances": 19,
              "style": Object {
                "fontSize": 30,
              },
              "text": "update",
            },
            "ref": null,
            "rendered": "update",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "readme",
            "nodeType": "function",
            "props": Object {
              "children": "readme",
              "instances": 9,
              "style": Object {
                "fontSize": 30,
              },
              "text": "readme",
            },
            "ref": null,
            "rendered": "readme",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "removes",
            "nodeType": "function",
            "props": Object {
              "children": "removes",
              "instances": 12,
              "style": Object {
                "fontSize": 30,
              },
              "text": "removes",
            },
            "ref": null,
            "rendered": "removes",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "vulnerability",
            "nodeType": "function",
            "props": Object {
              "children": "vulnerability",
              "instances": 1,
              "style": Object {
                "fontSize": 14,
                "opacity": 0.7,
              },
              "text": "vulnerability",
            },
            "ref": null,
            "rendered": "vulnerability",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "completes",
            "nodeType": "function",
            "props": Object {
              "children": "completes",
              "instances": 2,
              "style": Object {
                "fontSize": 14,
                "opacity": 0.7,
              },
              "text": "completes",
            },
            "ref": null,
            "rendered": "completes",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "admin",
            "nodeType": "function",
            "props": Object {
              "children": "admin",
              "instances": 7,
              "style": Object {
                "fontSize": 22,
              },
              "text": "admin",
            },
            "ref": null,
            "rendered": "admin",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "testing",
            "nodeType": "function",
            "props": Object {
              "children": "testing",
              "instances": 2,
              "style": Object {
                "fontSize": 14,
                "opacity": 0.7,
              },
              "text": "testing",
            },
            "ref": null,
            "rendered": "testing",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "activity_index_spec",
            "nodeType": "function",
            "props": Object {
              "children": "activity_index_spec",
              "instances": 1,
              "style": Object {
                "fontSize": 14,
                "opacity": 0.7,
              },
              "text": "activity_index_spec",
            },
            "ref": null,
            "rendered": "activity_index_spec",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "edit\\"",
            "nodeType": "function",
            "props": Object {
              "children": "edit\\"",
              "instances": 1,
              "style": Object {
                "fontSize": 14,
                "opacity": 0.7,
              },
              "text": "edit\\"",
            },
            "ref": null,
            "rendered": "edit\\"",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="tag-container"
        >
          <TagCloud
            className="tag-cloud"
            onClick={[Function]}
            random={[Function]}
            rotate={0}
            spiral="archimedean"
            style={
              Object {
                "color": [Function],
                "fontFamily": "sans-serif",
                "fontSize": 22,
                "height": "100%",
                "padding": 5,
                "rotate": 90,
                "width": "100%",
              }
            }
          >
            <CloudItem
              instances={19}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="update"
            >
              update
            </CloudItem>
            <CloudItem
              instances={9}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="readme"
            >
              readme
            </CloudItem>
            <CloudItem
              instances={12}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="removes"
            >
              removes
            </CloudItem>
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="vulnerability"
            >
              vulnerability
            </CloudItem>
            <CloudItem
              instances={2}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="completes"
            >
              completes
            </CloudItem>
            <CloudItem
              instances={7}
              style={
                Object {
                  "fontSize": 22,
                }
              }
              text="admin"
            >
              admin
            </CloudItem>
            <CloudItem
              instances={2}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="testing"
            >
              testing
            </CloudItem>
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="activity_index_spec"
            >
              activity_index_spec
            </CloudItem>
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="edit\\""
            >
              edit"
            </CloudItem>
          </TagCloud>
        </div>,
        "className": "commits-word-cloud",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TagCloud
            className="tag-cloud"
            onClick={[Function]}
            random={[Function]}
            rotate={0}
            spiral="archimedean"
            style={
              Object {
                "color": [Function],
                "fontFamily": "sans-serif",
                "fontSize": 22,
                "height": "100%",
                "padding": 5,
                "rotate": 90,
                "width": "100%",
              }
            }
          >
            <CloudItem
              instances={19}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="update"
            >
              update
            </CloudItem>
            <CloudItem
              instances={9}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="readme"
            >
              readme
            </CloudItem>
            <CloudItem
              instances={12}
              style={
                Object {
                  "fontSize": 30,
                }
              }
              text="removes"
            >
              removes
            </CloudItem>
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="vulnerability"
            >
              vulnerability
            </CloudItem>
            <CloudItem
              instances={2}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="completes"
            >
              completes
            </CloudItem>
            <CloudItem
              instances={7}
              style={
                Object {
                  "fontSize": 22,
                }
              }
              text="admin"
            >
              admin
            </CloudItem>
            <CloudItem
              instances={2}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="testing"
            >
              testing
            </CloudItem>
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="activity_index_spec"
            >
              activity_index_spec
            </CloudItem>
            <CloudItem
              instances={1}
              style={
                Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                }
              }
              text="edit\\""
            >
              edit"
            </CloudItem>
          </TagCloud>,
          "className": "tag-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <CloudItem
                instances={19}
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
                text="update"
              >
                update
              </CloudItem>,
              <CloudItem
                instances={9}
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
                text="readme"
              >
                readme
              </CloudItem>,
              <CloudItem
                instances={12}
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
                text="removes"
              >
                removes
              </CloudItem>,
              <CloudItem
                instances={1}
                style={
                  Object {
                    "fontSize": 14,
                    "opacity": 0.7,
                  }
                }
                text="vulnerability"
              >
                vulnerability
              </CloudItem>,
              <CloudItem
                instances={2}
                style={
                  Object {
                    "fontSize": 14,
                    "opacity": 0.7,
                  }
                }
                text="completes"
              >
                completes
              </CloudItem>,
              <CloudItem
                instances={7}
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
                text="admin"
              >
                admin
              </CloudItem>,
              <CloudItem
                instances={2}
                style={
                  Object {
                    "fontSize": 14,
                    "opacity": 0.7,
                  }
                }
                text="testing"
              >
                testing
              </CloudItem>,
              <CloudItem
                instances={1}
                style={
                  Object {
                    "fontSize": 14,
                    "opacity": 0.7,
                  }
                }
                text="activity_index_spec"
              >
                activity_index_spec
              </CloudItem>,
              <CloudItem
                instances={1}
                style={
                  Object {
                    "fontSize": 14,
                    "opacity": 0.7,
                  }
                }
                text="edit\\""
              >
                edit"
              </CloudItem>,
            ],
            "className": "tag-cloud",
            "onClick": [Function],
            "random": [Function],
            "rotate": 0,
            "spiral": "archimedean",
            "style": Object {
              "color": [Function],
              "fontFamily": "sans-serif",
              "fontSize": 22,
              "height": "100%",
              "padding": 5,
              "rotate": 90,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "update",
              "nodeType": "function",
              "props": Object {
                "children": "update",
                "instances": 19,
                "style": Object {
                  "fontSize": 30,
                },
                "text": "update",
              },
              "ref": null,
              "rendered": "update",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "readme",
              "nodeType": "function",
              "props": Object {
                "children": "readme",
                "instances": 9,
                "style": Object {
                  "fontSize": 30,
                },
                "text": "readme",
              },
              "ref": null,
              "rendered": "readme",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "removes",
              "nodeType": "function",
              "props": Object {
                "children": "removes",
                "instances": 12,
                "style": Object {
                  "fontSize": 30,
                },
                "text": "removes",
              },
              "ref": null,
              "rendered": "removes",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "vulnerability",
              "nodeType": "function",
              "props": Object {
                "children": "vulnerability",
                "instances": 1,
                "style": Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                },
                "text": "vulnerability",
              },
              "ref": null,
              "rendered": "vulnerability",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "completes",
              "nodeType": "function",
              "props": Object {
                "children": "completes",
                "instances": 2,
                "style": Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                },
                "text": "completes",
              },
              "ref": null,
              "rendered": "completes",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "admin",
              "nodeType": "function",
              "props": Object {
                "children": "admin",
                "instances": 7,
                "style": Object {
                  "fontSize": 22,
                },
                "text": "admin",
              },
              "ref": null,
              "rendered": "admin",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "testing",
              "nodeType": "function",
              "props": Object {
                "children": "testing",
                "instances": 2,
                "style": Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                },
                "text": "testing",
              },
              "ref": null,
              "rendered": "testing",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "activity_index_spec",
              "nodeType": "function",
              "props": Object {
                "children": "activity_index_spec",
                "instances": 1,
                "style": Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                },
                "text": "activity_index_spec",
              },
              "ref": null,
              "rendered": "activity_index_spec",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "edit\\"",
              "nodeType": "function",
              "props": Object {
                "children": "edit\\"",
                "instances": 1,
                "style": Object {
                  "fontSize": 14,
                  "opacity": 0.7,
                },
                "text": "edit\\"",
              },
              "ref": null,
              "rendered": "edit\\"",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
